sklearn/utils/estimator_checks.py

```diff
@@ -1443,41 +1443,87 @@ def check_supervised_y_2d(name, estimator_orig):
     assert_allclose(y_pred.ravel(), y_pred_2d.ravel())
 
 
-@ignore_warnings(category=(DeprecationWarning, FutureWarning))
+@ignore_warnings
+def check_classifiers_classes_helper(X, y, name, classifier_orig):
+    classes = np.unique(y)
+    classifier = clone(classifier_orig)
+    if name == 'BernoulliNB':
+        X = X > X.mean()
+    set_random_state(classifier)
+    # fit
+    classifier.fit(X, y)
+
+    y_pred = classifier.predict(X)
+
+    if hasattr(classifier, "decision_function"):
+        decision = classifier.decision_function(X)
+        n_samples, n_features = X.shape
+        msg_shape="Predicted shape of `decision_function` mismatches the expected " \
+                  "shape: expected '%s', got '%s'"
+        if len(classes) == 2:
+            assert_equal(decision.shape, (n_samples, ), msg_shape % \
+                        ((n_samples, ), decision.shape))
+            dec_pred = (decision.ravel() > 0).astype(np.int)
+            dec_exp = classifier.classes_[dec_pred]
+            msg="Predicted class mismatch for classifier %r: " \
+                "expected '%s', got '%s' " % \
+                (classifier, ", ".join(map(str, dec_exp)), ", ".join(map(str, y_pred)))
+            assert_array_equal(dec_exp, y_pred, err_msg=msg)
+        elif getattr(classifier, 'decision_function_shape', 'ovr') == 'ovr':
+            assert_equal(decision.shape, (n_samples, len(classes)), msg_shape % \
+                        ((n_samples, len(classes)), decision.shape))
+            decision_y = np.argmax(decision, axis=1).astype(int)
+            assert_array_equal(classifier.classes_[decision_y], y_pred, msg_shape % \
+                              ((n_samples, len(classes)), decision.shape))
+
+    # training set performance
+    if name != "ComplementNB":
+        # This is a pathological data set for ComplementNB.
+        assert_array_equal(np.unique(y), np.unique(y_pred))
+    if np.any(classifier.classes_ != classes):
+        msg="Unexpected classes_ attribute for %r: expected '%s', got '%s'" % \
```

Comment:
@jnothman , do you have a suggestion what I could use instead of backslash. With backslash it reports `E502 the backslash is redundant between brackets`

Reply:
Put the whole expression in parentheses and remove the backslash

Reply:
Thank you :)

Reply:
@jnothman , I tried different ways of putting parenthesis but  it still fails.
`assert_array_equal(y_exp, y_pred, err_msg=msg_classes
  % (classifier, ", ".join(map(str, y_exp)), ", ".join(map(str, y_pred))))
`
What kind of parentheses do you have in your mind ? Would you mind showing it on the above example ?

Reply:
I would probably do the following:
```py
        assert_array_equal(classes, classifier.classes_,
                           err_msg="Unexpected classes_ attribute for %r: "
                                   "expected '%s', got '%s'" %
                                   (classifier,
                                    ", ".join(map(str, classes)),
                                    ", ".join(map(str, classifier.classes_))))
```

Reply:
Thank you, it worked :)